cmd/clusterctl/client/cluster/objectgraph.go

```diff
@@ -410,8 +410,8 @@ func getCRDList(proxy Proxy, crdList *apiextensionsv1.CustomResourceDefinitionLi
 }
 
 // Discovery reads all the Kubernetes objects existing in a namespace (or in all namespaces if empty) for the types received in input, and then adds
-// everything to the objects graph.
-func (o *objectGraph) Discovery(namespace string) error {
+// everything to the objects graph. Filters for objects only belonging to specific cluster if provided.
+func (o *objectGraph) Discovery(namespace, cluster string) error {
```

Comment:
```suggestion
func (o *objectGraph) Discovery(namespace, clusterName string) error {
```
and in other places too 🙏🏻